src/context/AppProvider/AppContext.tsx

```diff
@@ -120,9 +129,23 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
         Object.assign(accountMetadataByAccountId, accountIdsAndMetadata)
 
         const { getAccount } = portfolioApi.endpoints
-        const accountPromises = accountIds.map(accountId =>
-          dispatch(getAccount.initiate({ accountId }, { forceRefetch: true })),
-        )
+
+        const accountPromises = accountIds.map(accountId => {
+          const accountChainId = fromAccountId(accountId).chainId
+          // Do not try and fetch "multi-accounts" for MetaMask snaps
+          // Snaps do not support using the wallet entropy for EVM chains and the native MM functionality is used instead
+          // This means we would get the same address for all account numbers and end in an infinite loop here
+          if (
+            accountNumber > 0 &&
+            // Cosmos SDK chains account derivation > 0 is rugged for snaps, wut?
+            (isEvmChainId(accountChainId) ||
+              fromChainId(accountChainId).chainNamespace === CHAIN_NAMESPACE.CosmosSdk) &&
+            isMetaMaskMultichainWallet
+          )
```

Comment:
Given this logic is relatively magic and duplicated in `<AddAccountModal />`, we may want to add a `isSupportedMultiAccountForChain` or similar util

Reply:
eda188bf374521d112061933db79d350e0f431de and mandatory jam retest https://jam.dev/c/6cf6fa70-5cd7-4324-8345-761f0d37a45e